service/grails-app/controllers/mod/rs/SettingController.groovy

```diff
@@ -132,7 +135,22 @@ class SettingController extends OkapiTenantAwareSwaggerController<AppSetting> {
         }
 
         // Now we can perform the lookup
-        respond doTheLookup(gormFilterClosure);
+        def result = doTheLookup(gormFilterClosure);
+
+        // Check if we need to filter out the 'combine_fill_and_ship' option based on feature flags
+        if (params.filters.contains(SettingsData.SECTION_STATE_ACTION_CONFIG)) {
+            // Retrieve the feature flag value for state action configuration
+            String featureFlag = settingsService.getSettingValue(SettingsData.SETTING_FEATURE_FLAG_STATE_ACTION_CONFIGURATION_COMBINE_FILL_AND_SHIP)
```

Comment:
@EddiTim we shouldn't harcode any particular feature flag here. Instead iterate through every record in the result and construct the feature flag setting like this:

```
String featFlag = record.section + "." + record.key + "." + "feature_flag"
```
then get the value of the featureFlag and if it's not null and "false", remove the setting